crates/provider/src/native.rs

```diff
@@ -0,0 +1,323 @@
+use std::error::Error;
+
+use crate::shared::{
+    client::Client,
+    types::{FileMap, NameSpaceDef, PodDef, RunCommandOptions, RunCommandResponse, Settings},
+};
+
+struct Native {
+    // Namespace of the client
+    namespace:             String,
+    // path where configuration relies
+    config_path:           String,
+    // variable that shows if debug is activated
+    debug:                 bool,
+    // the timeout for the client to exit
+    timeout:               u32,
+    // command sent to client
+    command:               String,
+    // temporary directory
+    tmp_dir:               String,
+    pod_monitor_available: bool,
+    local_magic_file_path: String,
+    // name of the provider
+    provider_name:         String,
+}
+
+impl Default for Native {
+    fn default() -> Self {
+        // [TODO]: define the default value for Native
+        todo!()
+    }
+}
+
+impl Native {
+    pub fn new(namespace: &str, config_path: &str, tmp_dir: &str) -> Native {
```

Comment:
```suggestion
    pub fn new(namespace: impl Into<String>, config_path: impl Into<String>, tmp_dir: impl Into<String>) -> Native {
```
Using `impl Into<String>` give us more flexibility.